Add guide for kubeconfig set (#1363)
run-ai/docs: docs/Researcher/cli-reference/new-cli/guides/set-kubeconfig-with-oidc-parameters.md

```
runai kubeconfig set
```

## 

## Prerequisites

Before executing the command, ensure that

1. Cluster authentication is configured and enabled.
2. The user has a kubeconfig file configured.
3. The user is logged in (use the [runai login](http://./runai_login.md)  command)

### Cluster configuration

To enable cluster authentication, add the following flags to the Kubernetes server API of each cluster

```
spec:
  containers:
  - command:
    ...
    - --oidc-client-id=<OIDC_CLIENT_ID>
    - --oidc-issuer-url=url=https://<HOST>/auth/realms/<REALM>
    - --oidc-username-prefix=-
```

### 

### User Kubeconfig configuration

Add the following to the Kubernetes client configuration file (./kube/config)

```
apiVersion: v1
kind: Config
preferences:
  colors: true
current-context: <CONTEXT_NAME>
contexts:
- context:
    cluster: <CLUSTER_NAME>
    user: <USER_NAME>
  name: <CONTEXT_NAME>
clusters:
- cluster:
    server: <CLUSTER_URL>
    certificate-authority-data: <CLUSTER_CERT>
  name: <CLUSTER_NAME>
users:
- name: <USER_NAME>
```

* Make sure to replace values with the actual cluster information and user credentials.  

> * There can be multiple contexts in the kubeconfig file, the command will config the current context

Author: ```suggestion
* There can be multiple contexts in the kubeconfig file. The command will configure the current context.
```

# Add Run:ai authorization to kubeconfig

> The ***runai kubeconfig set*** command allows users to configure their kubeconfig file with Run:AI authorization token. This setup enables users to gain access to the Kubernetes (k8s) cluster seamlessly.

Reviewer: Run:ai

> The ***runai kubeconfig set*** command allows users to configure their kubeconfig file with Run:AI authorization token. This setup enables users to gain access to the Kubernetes (k8s) cluster seamlessly.

Author: ```suggestion
The ***runai kubeconfig set*** command allows users to configure their kubeconfig file with Run:ai authorization token. This setup enables users to gain access to the Kubernetes (k8s) cluster seamlessly.
```

> * There is no need to set the kubeconfig to work with the CLI properly. It is used only to work with external workloads under Run:ai authorization.

Reviewer: Not sure what this means - what set up is required to use the CLI? Also external workloads meaning third-party workloads?

Author: yes 3rd party workloads. if you want to submit them, you need direct connection to the k8s cluster. in order to do so, you'll need to use the new command. but for all regular work with the cli you don't need it. 
This disclaimer comes to make sure users won't get confused and think they must use this command. it's only for advanced users

Reviewer: So let's change it to this:

!!!Note
   Setting kubeconfig is not required in order to use the CLI. This command is used to enable third-party workloads under Run:ai authorization.

> * There is no need to set the kubeconfig to work with the CLI properly. It is used only to work with external workloads under Run:ai authorization.

Author: ```suggestion
!!!Note
Setting kubeconfig is not required in order to use the CLI. This command is used to enable third-party workloads under Run:ai authorization.
```

## Usage

> To set the token inside the kubeconfig file, run the following command:

Reviewer: Where do they obtain the token?

Author: run:ai automatically fetch and set it for you

Reviewer: At what stage does it fetch the token? Let's add this somewhere.

> To set the token inside the kubeconfig file, run the following command:

Author: ```suggestion
To set the token (will be fetched automatically, no need to bring it manually) inside the kubeconfig file, run the following command:
```

Reviewer: To set the token (fetched automatically) inside the kubeconfig file, run the following command:

```
runai kubeconfig set
```

## 

## Prerequisites

Before executing the command, ensure that

1. Cluster authentication is configured and enabled  

> 2. The user has a kubeconfig file that includes the following elements

Reviewer: Which elements?

> 3. The user is logged in (use the 'runai login' command)

Reviewer: let's add a reference to this command

> ### 

Reviewer: remove

Author: remove what?

Reviewer: The ###

> ### 

Author: ```suggestion

```

### Cluster configuration

To enable cluster authentication, add the following flags to the Kubernetes server API of each cluster

```
spec:
  containers:
  - command:
    ...
    - --oidc-client-id=<OIDC_CLIENT_ID>
    - --oidc-issuer-url=url=https://<HOST>/auth/realms/<REALM>
    - --oidc-username-prefix=-
```

### 

### User Kubeconfig configuration

Add the following to the Kubernetes client configuration file (./kube/config)

```
apiVersion: v1
kind: Config
preferences:
  colors: true
current-context: <CONTEXT_NAME>
contexts:
- context:
    cluster: <CLUSTER_NAME>
    user: <USER_NAME>
  name: <CONTEXT_NAME>
clusters:
- cluster:
    server: <CLUSTER_URL>
    certificate-authority-data: <CLUSTER_CERT>
  name: <CLUSTER_NAME>
users:
- name: <USER_NAME>
```

* Make sure to replace values with the actual cluster information and user credentials.  

> * There can be multiple contexts in the kubeconfig file, the command will config the current context

Reviewer: * There can be multiple contexts in the kubeconfig file. The command will configure the current context.

> * There can be multiple contexts in the kubeconfig file, the command will config the current context

Author: ```suggestion
* There can be multiple contexts in the kubeconfig file. The command will config the current context.
```

> For the full command reference see [here](http://./guides/runai_kubeconfig_set.md)  

Reviewer: For the full command reference, see [kubeconfig set](http://./guides/runai_kubeconfig_set.md). 
Is this the correct path?

Author: i hope, not sure how to  test it. the mkdocs.yaml change should sort it
